lib/encoder.py

```diff
@@ -796,15 +801,31 @@ def StoreEncoder(self, encoder, workdir=None):
       parameterfile.write(encoder.parameters.ToString())
     encoder.stored = True
 
-  def ReadEncoderParameters(self, hashname):
-    dirname = os.path.join(self.workdir, hashname)
-    with open(os.path.join(dirname, 'parameters'), 'r') as parameterfile:
-      return OptionValueSet(self.context.codec.option_set, parameterfile.read(),
-                            formatter=self.context.codec.option_formatter)
-
-  def AllEncoderFilenames(self):
-    pattern = os.path.join(self.workdir, '*')
-    return [os.path.basename(this_file) for this_file in glob.glob(pattern)]
+  def ReadEncoderParameters(self, dirname):
+    if dirname[0:0] == '/':
+      with open(os.path.join(dirname, 'parameters'), 'r') as parameterfile:
+        return OptionValueSet(self.context.codec.option_set,
+                              parameterfile.read(),
+                              formatter=self.context.codec.option_formatter)
+    else:
+      for repository in self.SearchPath():
+        filename = os.path.join(repository, dirname, 'parameters')
+        if os.path.isfile(filename):
+          with open(filename, 'r') as parameterfile:
+            return OptionValueSet(self.context.codec.option_set,
+                                  parameterfile.read(),
+                                  formatter=self.context.codec.option_formatter)
+
+  def AllEncoderFilenames(self, only_workdir=False):
```

Comment:
Consider flipping this to also_include_imported=True. I think that gives a bit more information to the reader since the workdir is used more everywhere, but it's not immediately clear what we are adding besides the workdir in the above.